modules/internal/BytesStringCommon.chpl

```diff
@@ -1426,6 +1426,9 @@ module BytesStringCommon {
         }
       }
     }
+
+    // ensure that there is a null byte at the end of the buffer
+    x.buff[x.buffLen] = 0;
```

Comment:
Recommend `if x.buffLen > 0` here around this, unless you know that `x` cannot be the empty string at this point.